### DISCUSSION TITLE
How to Add a Custom Model in PaddleOCR（如何新增自定义模型）？

### DISCUSSION THREAD

**Discussion by RebornForPower** [PaddlePaddle/PaddleOCR#14276] [Q&A]
I have noticed that while the official documentation for PaddleOCR provides detailed instructions on the application of existing models in the model library, it lacks sufficient guidance on how to train and add custom models. I am looking for relevant tutorials or examples to help smoothly train my own model and integrate it into PaddleOCR.

For instance, I would like to integrate the latest YOLO model directly into PaddleOCR. Rather than using the YOLO framework separately along with PaddleOCR (which often leads to numerous bugs and significant overhead), I want to directly embed the YOLO model into PaddleOCR to streamline the process and optimize performance.

我在使用PaddleOCR的过程中发现，官方文档中对模型库中现有模型的应用有详细的说明，但对于如何训练和新增自定义模型的指导相对缺乏。我希望能找到相关的教程或范例，以便顺利训练自己的模型并将其整合进PaddleOCR。

例如，我想将最新的YOLO模型直接整合到PaddleOCR中。与其像网上那样单独使用YOLO框架和PaddleOCR框架（这种做法通常会导致很多bug并造成较大的开销），我希望能够直接将YOLO模型嵌入到PaddleOCR中，以简化流程并优化性能。这种集成方式能让我更高效地进行训练和推理，同时减少潜在的问题

**Comment by GreatV** [comment]
This is a very complex task. I hope the following link can provide you with a reference: https://paddlepaddle.github.io/PaddleOCR/latest/en/algorithm/add_new_algorithm.html. As for YOLO, take a look at [PaddleYOLO](https://github.com/PaddlePaddle/PaddleYOLO).
